spark-plugins/src/test/java/co/cask/hydrator/plugin/spark/test/NGramTransformTest.java

```diff
@@ -0,0 +1,253 @@
+                                             String mockNameOfSinkPlugin, String ngramSize) {
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(new ETLStage("source", MockSource.getPlugin(mockNameOfSourcePlugin)))
+      .addStage(new ETLStage("sparkcompute",
+                             new ETLPlugin(NGramTransform.PLUGIN_NAME, SparkCompute.PLUGIN_TYPE,
+                                           ImmutableMap.of("outputField", OUTPUT_FIELD,
+                                                           "fieldToBeTransformed", FIELD_TO_BE_TRANSFORMED,
+                                                           "ngramSize", ngramSize),
+                                           null))).addStage(new ETLStage("sink",
+                                                                         MockSink.getPlugin(mockNameOfSinkPlugin)))
+      .addConnection("source", "sparkcompute")
+      .addConnection("sparkcompute", "sink")
+      .build();
+    return etlConfig;
+  }
+
+  @Test
+  public void testMultiFieldsSourceWith2N() throws Exception {
+    String mockSource = "Multiple";
+    String multiFieldData = "MultipleFields";
+    /*
+     * source --> sparkcompute --> sink
+    */
+    ETLBatchConfig etlConfig = buildETLBatchConfig(mockSource, multiFieldData, "2");
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app(APP_NAME);
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+    DataSetManager<Table> inputManager = getDataset(mockSource);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(SOURCE_SCHEMA_MULTIPLE).set(FIELD_TO_BE_TRANSFORMED,
+                                                           DATA_TO_BE_TRANSFORMED1)
+                                                      .set(ADDITIONAL_FIELD, "CDAP1").build(),
+      StructuredRecord.builder(SOURCE_SCHEMA_MULTIPLE).set(FIELD_TO_BE_TRANSFORMED,
+                                                           DATA_TO_BE_TRANSFORMED2)
+                                                      .set(ADDITIONAL_FIELD, "CDAP2").build()
+    );
+    MockSource.writeInput(inputManager, input);
+    // manually trigger the pipeline
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+    DataSetManager<Table> nGrams = getDataset(multiFieldData);
+    List<StructuredRecord> output = MockSink.readOutput(nGrams);
+    Set<List> results = new HashSet<>();
+    for (StructuredRecord structuredRecord : output) {
+      results.add((ArrayList) structuredRecord.get(OUTPUT_FIELD));
+    }
+    StructuredRecord row = output.get(0);
+    Assert.assertEquals(2, output.size());
+    Assert.assertEquals(getExpectedData(), results);
+    Assert.assertTrue(getExpectedData().containsAll(results));
```

Comment:
I think `Assert.assertEquals(getExpectedData(), results);` will check for both the size and contain. so you dont actually need the `containsAll` assert.